gradle-plugin/plugin/src/main/java/com/yelp/codegen/plugin/GenerateTask.kt

```diff
@@ -83,13 +92,13 @@ open class GenerateTask : DefaultTask() {
             specName ${'\t'} ${specName ?: "[ DEFAULT ] $DEFAULT_NAME"}
             specVers ${'\t'} ${specVersion ?: "[ DEFAULT ] $DEFAULT_VERSION"}
             input ${"\t\t"} $inputFile
-            output ${"\t\t"} ${outputDir ?: "[ DEFAULT ] $defaultOutputDir"}
+            output ${"\t\t"} ${outputDirectory ?: "[ DEFAULT ] $defaultOutputDir"}
             groupId ${'\t'} ${packageName ?: "[ DEFAULT ] default"}
             artifactId ${'\t'} ${packageName ?: "[ DEFAULT ] com.codegen"}
-            features ${'\t'} ${features?.headersToRemove?.joinToString(", ")?.ifEmpty { "[  EMPTY  ]" }}
+            features ${'\t'} ${headersToRemove.joinToString(separator = ",", prefix = "[", postfix = "]")}
         """.trimIndent())
 
-        val packageName = packageName ?: DEFAULT_PACKAGE
+        packageName = this.packageName.orNull ?: DEFAULT_PACKAGE
```

Comment:
another cool part of the property api's in gradle is the ability to have a `convention` on them.

helps keep things more declarative and removes the need to check for null and such later on.
https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Property.html#convention-T-

there is both a `convention(T)` and `convention(Provider<T>)` so if it is something that needs to be lazily calculated you can still do that at a higher level and keep this task execution code more streamlined.

Reply:
That looks great! 

@martinbonnin would you be able to integrate this suggestion?

Reply:
Not sure how conventions would help here as we overwrite the task properties from the `GradleTaskConfiguration` object and this will remove any previously set convention.

@trevjonez did you have something specific in mind ?

Reply:
In that case you could set the defaults as the convention on the property of the gradle task configuration object. 

Looking at this more, I think most if not all of the properties on the task should not be marked optional since a value is required for the task to run. My thinking is that not having the defaults in the property when the task is checked before running it might open you up to odd build cache key issues.

Reply:
Yea, good point. That's a bigger change but I guess now is the time :). I'll look into it.

Reply:
Changes done, let me know what you think.